librz/core/basefind.c

```diff
@@ -35,14 +40,50 @@ typedef struct basefind_thread_data_t {
 	RzList *scores;
 	HtUU *pointers;
 	BaseFindArray *array;
+	ThreadBool loop;
 } BaseFindThreadData;
 
 typedef struct basefind_ui_info_t {
+	ThreadBool loop;
 	RzThreadPool *pool;
 	void *user;
 	RzBaseFindThreadInfoCb callback;
 } BaseFindUIInfo;
 
+static void thread_bool_init(ThreadBool *tl) {
+	tl->lock = rz_th_lock_new(false);
+	tl->value = true;
+}
+
+static bool thread_bool_get(ThreadBool *tl) {
+	rz_th_lock_enter(tl->lock);
+	bool value = tl->value;
+	rz_th_lock_leave(tl->lock);
+	return value;
+}
+
+static void thread_bool_set(ThreadBool *tl, bool value) {
+	rz_th_lock_enter(tl->lock);
+	tl->value = value;
+	rz_th_lock_leave(tl->lock);
+}
+
+static void thread_bool_fini(ThreadBool *tl) {
+	rz_th_lock_free(tl->lock);
+}
+
```

Comment:
I'm not sure about this TBH... Is all this thing really necessary for a bool? I think there are some functions like cmp&xchg or something like that that can be used to atomically set a bool value.

Reply:
This is what is what happens when you do multithreaded code, becomes very weird.